File: LeetCode/problems/70-climbing-stairs/70-climbing-stairs.py
##int climbStairs(int n) {
 ##   if(n <= 3){return n;}
 ##   return climbStairs(n-1)+climbStairs(n-2);
##}
class Solution:
    @cache
    def climbStairs(self, n: int) -> int:
        if n <= 3:
            return n
        return self.climbStairs(n-1)+self.climbStairs(n-2)

File: LeetCode/problems/70-climbing-stairs/70-climbing-stairs.php
class Solution {
    private array $memo = [0,1,2,3];
    function climbStairs($n) {
        if($n <= 3){return $n;}

        if($this->$memo[$n] == NULL){
            $this->$memo[$n] = $this->climbStairs($n-1)+$this->climbStairs($n-2);
        }

        return $this->$memo[$n];
    }
}